documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -3,118 +3,158 @@
+
+== Uploading the demo realm
+
+When a Keycloak instance is running, upload the example realm to Keycloak using the Admin Console.
 
 
-== Starting the pods
+First, determine the external hostname to access the Admin Console.
 
-For this example, we use the kubernetes deployment scripts available at {oauth-kubernetes-examples-link}:
+For example, if the Keycloak Operator installed the Keycloak server in the `sso` namespace, run the following command:
 
-[source,yaml,subs="attributes"]
+[source,shell,subs="attributes"]
 ----
-export ROOT=https://raw.githubusercontent.com/strimzi/strimzi-kafka-oauth/{oauth-version}
+NS=sso
+kubectl get ingress keycloak -n $NS
 ----
 
-.Deploy the Postgres database for Keycloak
+Second, get the password for the admin user.
+
+If you used Keycloak Operator, the password is stored as a secret, so you first have to identify the name of the secret. You can inspect the yaml for the Keycloak instance. For example:
 
-[source,yaml]
+[source,shell,subs="attributes"]
 ----
-kubectl apply -f $ROOT/examples/kubernetes/postgres-pvc.yaml
-kubectl apply -f $ROOT/examples/kubernetes/postgres.yaml
+kubectl get -n $NS pod keycloak-0 -o yaml | less
 ----
 
-.Deploy the Keycloak server
+Search for `KEYCLOAK_PASSWORD` and note the `secretKeyRef`.`name` (usually `credential-keycloak` or similar).
 
-[source,yaml]
+You can then obtain the clear text password by running:
+
+[source,shell,subs="attributes"]
 ----
-kubectl apply -f $ROOT/examples/kubernetes/keycloak-realms-configmap.yaml
-kubectl apply -f $ROOT/examples/kubernetes/keycloak-postgres.yaml
+SECRET_NAME=credential-keycloak
+kubectl get -n $NS secret $SECRET_NAME -o yaml | grep PASSWORD | awk '{print $2}' | base64 -D
 ----
 
-If your default namespace is not `myproject` (for example, `default`), use the following to deploy Keycloak:
+Log in to the Admin Console with the username `admin` and the password you obtained earlier. You must use `https://HOST` to access the Kubernetes ingress.
+
+When you are logged in, click *Add Realm* to import the example realm.
+
+Select the `$EXAMPLES_DIR/security/keycloak-authorization/kafka-authz-realm.json` file from your disk, and then click *Create*.
+
+You should now have `kafka-authz` as your current realm in the Admin Console.
+
+== Deploying the minimal Kafka cluster
 
-[source,yaml]
+We assume that the Strimzi Cluster Operator has already been installed on the Kubernetes  cluster.
+
+Our Kafka cluster has to be properly configured with authentication and authorization settings to connect to our Keycloak instance.
+
+Assuming the Keycloak Operator was used to install the Keycloak server instance, and that `sso` namespace was used, we can determine the external hostname:
+
+[source,shell,subs="attributes"]
 ----
-curl -s $ROOT/examples/kubernetes/keycloak-postgres.yaml | sed -e "s#myproject#default#" | kubectl apply -f -
+NS=sso
+kubectl get ingress keycloak -n $NS
```

Comment:
There seems to be a lot of inconsistency in the way you use the `$NS`: 
* In a lot of examples, you define it and use it
* In other, you just use it without defining it

I wonder if we can do it just one way. Not a big deal, you can keep it as it is if you see some structure in it. It is just something I noticed.